hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/DiscriminatorPathInterpretation.java

```diff
@@ -48,8 +48,6 @@ public DiscriminatorPathInterpretation(
 	public static SqmPathInterpretation<?> from(
 			DiscriminatorSqmPath path,
 			SqmToSqlAstConverter converter) {
-		assert path.getEntityDescriptor().hasSubclasses();
```

Comment:
Removing it looks fishy?

Reply:
Yeah. I'd think that we want to protect ourselves from creating the path interpretation if the entity type has no subtypes, because in that case we could have just created a literal directly instead. Unless you use the same code for any mappings now, I think the assert should stay there.

Reply:
It's another spot that limits the full consolidation of handling for entity and any discriminators.  That check has no meaning (its not even valid) for any discriminators.  So sure, I could add it back.  Of course, callers could also do that.

As entity and any discriminator handling is not currently unified here, that's less of a short-term concern.  But moving forward, it becomes a thing.

Reply:
I went ahead and added this assertion back.  As an illustration of what I am talking about, here is what that needs to look like now:

```
		assert path instanceof EntityDiscriminatorSqmPath;
		assert ( (EntityDiscriminatorSqmPath) path ).getEntityDescriptor().hasSubclasses();
```